### PR TITLE
Adding FS TC 83583721

### PR DESCRIPTION
**Pull request by AmarnatReddy**
# Description
Adding FS TC 83583721
CEPH-83583721 - Validate the details of all MDS servers are displayed with command `ceph mds metadata`.

        Steps:
        1. Create Filesystem with 2 active and 1 standby
        2. validate mds metadata info
        3. Make 1 active and 2 standby and enable standby reply
        4. validate mds metadata info
        5. Stop active MDS and Validate metadata info
        6. Stop Standby reply MDs and Validate metadata info

Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-2XCQ50/

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephfs/cephfs_bugs/validate_mds_metadata.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        log.info("Scenario 2: Validate standby replay MDS and check metadata")
+        client1.exec_command(
+            sudo=True, cmd=f"ceph fs set {fs_name} allow_standby_replay 1"
+        )
+        fs_util.wait_for_standby_replay_mds(client1, fs_name)
+        log.info("Scenario 2: Validate standby replay MDS and check metadata")
+        validate_metadata_fs_status(client1, fs_name)
+
+        log.info("Scenario 3: Fail Active MDS and check metadata")
+        fs_util.get_active_mdss(client1, fs_name=fs_name)
```
Comment (neha-gangadhar) [inline tests/cephfs/cephfs_bugs/validate_mds_metadata.py:85]: Do we need this function call get_active_mdss here? As we are executing same in the line 86 as well to capture the results.

File: tests/cephfs/cephfs_bugs/validate_mds_metadata.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        mds_ls = fs_util.get_active_mdss(client1, fs_name=fs_name)
+
+        mds_active_node = ceph_cluster.get_node_by_hostname(mds_ls[0].split(".")[1])
+        service_name = fs_util.deamon_name(mds_active_node, service=fs_name)
+        fs_util.deamon_op(
+            mds_active_node, service_name=service_name, op="stop", service=fs_name
+        )
+        log.info("Scenario 3: Fail Active MDS and check metadata")
```
Comment (neha-gangadhar) [inline tests/cephfs/cephfs_bugs/validate_mds_metadata.py:93]: May be the scenario logger needs to be modified as the step below is bringing back the Active MDS and validating metadata right? please check.

**Review by neha-gangadhar — COMMENTED** [review]
lgtm, minor comments. With one more approval can be merged.

File: suites/reef/cephfs/tier-3_cephfs_bugs.yaml [hunk 3]
```diff
@@ -18,6 +18,7 @@
   # CEPH-83583762: Check to prevent usage of Pools which has Pool level snaps for Filesystem creation.
   # CEPH-83584084: Validate if no repeated mclientscaps(revoke) entries found in MDS.
   # CEPH-83586730: validate standby replay node not removed after setting mds_inject_health_dummy
+  # CEPH-83583721 - Validate the details of all MDS servers are displayed with command `ceph mds metadata`.
```
Comment (hkadam134) [inline suites/reef/cephfs/tier-3_cephfs_bugs.yaml:21]: Can you include # CEPH-83583721 also in this ?

File: suites/reef/cephfs/tier-3_cephfs_bugs.yaml [hunk 4]
```diff
@@ -231,3 +232,9 @@ tests:
       polarion-id: CEPH-83586730
       desc: validate standby reply node not removed after setting mds_inject_health_dummy
       abort-on-fail: false
+  - test:
+      name:  Validate the details of all MDS servers are displayed with command ceph mds metadata
+      module: cephfs_bugs.validate_mds_metadata.py
+      polarion-id: CEPH-83583721
+      desc: Validate the details of all MDS servers are displayed with command ceph mds metadata
+      abort-on-fail: false
```
Comment (hkadam134) [inline suites/reef/cephfs/tier-3_cephfs_bugs.yaml:240]: Please add TC - #CEPH-83583721 test module in this file as the suite file was missing in previous PR

File: tests/cephfs/cephfs_bugs/validate_mds_metadata.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+            f"dd if=/dev/urandom of=subdir/sixmegs bs=1M conv=fdatasync count=6 seek=0;",
+        )
+        log.info("Scenario 1: Validate Active MDS and check metadata")
+        validate_metadata_fs_status(
+            client1,
+            fs_name,
+        )
+
+        log.info("set allow_standby_replay to true ")
+        client1.exec_command(sudo=True, cmd=f"ceph fs set {fs_name} max_mds 1")
```
Comment (hkadam134) [inline tests/cephfs/cephfs_bugs/validate_mds_metadata.py:75]: Can we validate the metadata here after setting the max_mds ?

Comment (AmarnatReddy) [reply tests/cephfs/cephfs_bugs/validate_mds_metadata.py:75]: In ceph mds metadata, Their was no mention of active or standby for mds nodes

**Review by hkadam134 — CHANGES_REQUESTED** [review]
Few minor changes requested, rest looks good.

File: tests/cephfs/cephfs_bugs/validate_mds_metadata.py (hunk 5 above)
Comment (hkadam134) [reply tests/cephfs/cephfs_bugs/validate_mds_metadata.py:75]: That's okay,  just to cover all possible scenarios we can include this too.

Comment (AmarnatReddy) [reply tests/cephfs/cephfs_bugs/validate_mds_metadata.py:75]: Added

File: suites/reef/cephfs/tier-3_cephfs_bugs.yaml (hunk 4 above)
Comment (AmarnatReddy) [reply suites/reef/cephfs/tier-3_cephfs_bugs.yaml:240]: Updated the suite file

File: suites/reef/cephfs/tier-3_cephfs_bugs.yaml (hunk 3 above)
Comment (AmarnatReddy) [reply suites/reef/cephfs/tier-3_cephfs_bugs.yaml:21]: updated the suite file

**Review by neha-gangadhar — APPROVED** [review]
Approved

File: suites/reef/cephfs/tier-3_cephfs_bugs.yaml [hunk 6]
```diff
@@ -231,3 +233,15 @@ tests:
...
+  - test:
+      name: Ensure deletion of clones are allowed when the clones are stuck and in pending state
+      module: cephfs_bugs.test_clone_delete_with_FS_fail.py
```
Comment (sumabai) [inline suites/reef/cephfs/tier-3_cephfs_bugs.yaml:244]: is this test already checked in and we are just updating now it suite?

**Review by hkadam134 — APPROVED** [review]
LGTM

**Pull request merged**